src/magellan_core/scripts/fake_sensors.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+import rospy
+from geometry_msgs.msg import Point, PolygonStamped
+from magellan_core.msg import (ObstacleStamped, ObstacleStampedArray)
+from visualization_msgs.msg import Marker, MarkerArray
+
+class FakeObstacles(object):
+    def __init__(self, obstacles):
+        self._marker_pub = rospy.Publisher('/obstacle_markers',
+                              MarkerArray,
+                              queue_size=5)
+
+        self._obst_pub = rospy.Publisher('/obstacles',
+                              ObstacleStampedArray,
+                              queue_size=5)
+
+        self._obstacle_msg = ObstacleStampedArray()
+        self._marker_message = MarkerArray()
+
+        marker_id = 1
+
+        for name, values in obstacles.iteritems():
+            # generate obstacle message
+            _obst = ObstacleStamped()
+            _obst.obst.x = values['x']
+            _obst.obst.y = values['y']
+            _obst.obst.width = values['width']
+            _obst.obst.length = values['length']
+            _obst.header.stamp = rospy.Time.now()
+            _obst.header.frame_id = 'map'
+
+            self._obstacle_msg.obstacles.append(_obst)
+
+            # generate marker message
+            marker = Marker()
+            marker.header.stamp = rospy.Time.now()
+            marker.header.frame_id = 'map'
+            marker.id = marker_id
+            marker.ns = 'obstacles'
+
+            marker.type = Marker.POINTS
+            marker.action = Marker.ADD
+            marker.pose.orientation.w = 1.0
+
+            marker.scale.x = 0.1
+            marker.scale.y = 0.1
+            marker.scale.z = 0.1
+
+            marker.color.r = 1.0
+            marker.color.g = 0.0
+            marker.color.b = 0.0
+            marker.color.a = 1.0
+
+            marker.lifetime = rospy.Duration(0.1)
+
+            # generate 4 corners of obstacles
+            center = Point()
+            center.x = _obst.obst.x
```

Comment:
I think the pairs generated for a rectangle with origin (x,y) being the top left and (x+width, y+length) being bottom right should be:

(x,y)
(x+width, y)
(x+width, y+width)
(x, y+width)

Reply:
fixed this